src/game/gui/textinput.c

```diff
@@ -52,16 +53,16 @@ static void textinput_render(component *c) {
             int xspace = c->w - tb->tconf.padding.left - tb->tconf.padding.right;
             start_x += ceilf((xspace - tmp_s) / 2.0f);
             tb->tconf.halign = TEXT_LEFT;
-            text_render(&tb->tconf, TEXT_DEFAULT, start_x + offset, c->y, c->w, c->h, "\x7F");
+            text_render(&tb->text_cache[0], &tb->tconf, TEXT_DEFAULT, start_x + offset, c->y, c->w, c->h, "\x7F");
             tb->tconf.halign = TEXT_CENTER;
         } else {
-            text_render(&tb->tconf, TEXT_DEFAULT, start_x + offset, c->y, c->w, c->h, "\x7F");
+            text_render(&tb->text_cache[1], &tb->tconf, TEXT_DEFAULT, start_x + offset, c->y, c->w, c->h, "\x7F");
```

Comment:
This can also use `text_cache[0]`, as it is an else.

Reply:
I agree. it looks like the default settings are never changed per object.